pymc3/variational/advi_minibatch.py

```diff
@@ -334,30 +477,22 @@ def get_transformed(v):
     )
 
     # Get global variables
-    global_RVs = list(set(vars) - set(list(local_RVs) + list(observed_RVs)))
-
-    # Ordering for concatenation of random variables
-    global_order = pm.ArrayOrdering([v for v in global_RVs])
-    local_order = pm.ArrayOrdering([v for v in local_RVs])
+    grvs = list(set(vars) - set(list(local_RVs) + list(observed_RVs)))
+    if global_RVs is None:
+        global_RVs = OrderedDict({v: 1 for v in grvs})
+    else:
+        assert(len(grvs) == len(global_RVs))
 
     # ELBO wrt variational parameters
-    inarray_g, uw_g, replace_g = _join_global_RVs(global_RVs, global_order)
-    inarray_l, uw_l, replace_l = _join_local_RVs(local_RVs, local_order)
-    logpt = _make_logpt(global_RVs, local_RVs, observed_RVs, model)
-    replace = replace_g
-    replace.update(replace_l)
-    logp = theano.clone(logpt, replace, strict=False)
-    elbo = _elbo_t(logp, uw_g, uw_l, inarray_g, inarray_l,
-                   n_mcsamples, random_seed)
-    del logpt
+    elbo, uw_l, uw_g = _make_elbo_t(observed_RVs, global_RVs, local_RVs, 
+                                    model.potentials, n_mcsamples, random_seed)
 
     # Replacements tensors of variational parameters in the graph
     replaces = dict()
 
     # Variational parameters for global RVs
     if 0 < len(global_RVs):
-        uw_global_shared, bij = _init_uw_global_shared(start, global_RVs,
-                                                       global_order)
+        uw_global_shared, bij = _init_uw_global_shared(start, global_RVs)
         replaces.update({uw_g: uw_global_shared})
 
     # Variational parameters for local RVs, encoded from samples in
```

Comment:
BTW this fails if n < 10. That number can be used in debugging 
```python
...
     for i in progress:
         e = f(*next(minibatches))
         elbos[i] = e
         if i % (n // 10) == 0 and i > 0:
...
```
PS the lowest line that was available for coment